code/modules/vending/_vending.dm

```diff
@@ -830,6 +859,65 @@ GLOBAL_LIST_EMPTY(vending_products)
 			R.amount--
 			SSblackbox.record_feedback("nested tally", "vending_machine_usage", 1, list("[type]", "[R.product_path]"))
 			vend_ready = TRUE
+		if("dispense")
```

Comment:
The other action being "vend" and this one being "dispense" are basically synonyms in this context and does not clearly indicate that this one is for custom products and the other is for default products

Reply:
ngl I only used that because it was [already present](https://github.com/yogstation13/Yogstation/blob/master/tgui/packages/tgui/interfaces/Vending.js#L69) on the tgui side (this whole thing was partially implemented):
```js
onClick={() => act('dispense', {
  'item': product.name,
})
```
but i'm happy to update it. while i'm at it, maybe i should refactor out the checking and price code into their own procs, to reduce the amount of duplicated code?